### PR TITLE
[Part Two] Reduced ResNet Model

### PR DESCRIPTION
**Pull request by jaku-jaku**
- Use Reduced ResNet Model to 4.9M instead of 22M parameters.
- Post homography operation to keep consistency 
- More features implemented for switching datasets, supporting different dataset folder arch
- Automation
- Adjusted batch size to 100 for RTX3090

### CODE REVIEW

**Comment by JXproject** [comment]
[2022-01-04 00:02:10.779420]: > epoch 78/200:
[2022-01-04 00:02:10.779448]:   >> Learning (wip) 
[2022-01-04 00:03:11.580097]:   >> Testing (wip) 
[2022-01-04 00:03:12.516874]:     epoch 78 > Training: [LOSS: -0.9848 | ACC: 0.9863] | Testing: [LOSS: -0.9840 | ACC: 0.9875] Ellapsed: 60.80 s | rate:0.93653

[2022-01-04 00:03:12.533398]: > Found Best Model State Dict saved @/home/jx/JX_Project/covid-xray-detection/output/CUSTOM-MODEL/latest-v8-reduced-model/models/best_state_dict_78:200.pth 

[2022-01-04 00:03:12.609813]: Best Classification Report:
----------------------
[2022-01-04 00:03:12.609855]:               precision    recall  f1-score   support

    positive       0.98      1.00      0.99       200
    negative       1.00      0.97      0.99       200

    accuracy                           0.99       400
   macro avg       0.99      0.99      0.99       400
weighted avg       0.99      0.99      0.99       400

**Comment by JXproject** [comment]
Layer (type)               Output Shape         Param #
================================================================
            Conv2d-1         [-1, 64, 160, 160]           9,408
       BatchNorm2d-2         [-1, 64, 160, 160]             128
              ReLU-3         [-1, 64, 160, 160]               0
         MaxPool2d-4           [-1, 64, 80, 80]               0
            Conv2d-5           [-1, 64, 80, 80]          36,864
       BatchNorm2d-6           [-1, 64, 80, 80]             128
              ReLU-7           [-1, 64, 80, 80]               0
            Conv2d-8           [-1, 64, 80, 80]          36,864
       BatchNorm2d-9           [-1, 64, 80, 80]             128
             ReLU-10           [-1, 64, 80, 80]               0
       BasicBlock-11           [-1, 64, 80, 80]               0
           Conv2d-12          [-1, 128, 40, 40]          73,728
      BatchNorm2d-13          [-1, 128, 40, 40]             256
             ReLU-14          [-1, 128, 40, 40]               0
           Conv2d-15          [-1, 128, 40, 40]         147,456
      BatchNorm2d-16          [-1, 128, 40, 40]             256
           Conv2d-17          [-1, 128, 40, 40]           8,192
      BatchNorm2d-18          [-1, 128, 40, 40]             256
             ReLU-19          [-1, 128, 40, 40]               0
       BasicBlock-20          [-1, 128, 40, 40]               0
           Conv2d-21          [-1, 256, 20, 20]         294,912
      BatchNorm2d-22          [-1, 256, 20, 20]             512
             ReLU-23          [-1, 256, 20, 20]               0
           Conv2d-24          [-1, 256, 20, 20]         589,824
      BatchNorm2d-25          [-1, 256, 20, 20]             512
           Conv2d-26          [-1, 256, 20, 20]          32,768
      BatchNorm2d-27          [-1, 256, 20, 20]             512
             ReLU-28          [-1, 256, 20, 20]               0
       BasicBlock-29          [-1, 256, 20, 20]               0
           Conv2d-30          [-1, 512, 10, 10]       1,179,648
      BatchNorm2d-31          [-1, 512, 10, 10]           1,024
             ReLU-32          [-1, 512, 10, 10]               0
           Conv2d-33          [-1, 512, 10, 10]       2,359,296
      BatchNorm2d-34          [-1, 512, 10, 10]           1,024
           Conv2d-35          [-1, 512, 10, 10]         131,072
      BatchNorm2d-36          [-1, 512, 10, 10]           1,024
             ReLU-37          [-1, 512, 10, 10]               0
       BasicBlock-38          [-1, 512, 10, 10]               0
AdaptiveAvgPool2d-39            [-1, 512, 1, 1]               0
           Linear-40                    [-1, 2]           1,026
           ResNet-41                    [-1, 2]               0
          Softmax-42                    [-1, 2]               0
================================================================
Total params: 4,906,818
Trainable params: 4,906,818
Non-trainable params: 0
----------------------------------------------------------------
Input size (MB): 1.17
Forward/backward pass size (MB): 87.11
Params size (MB): 18.72
Estimated Total Size (MB): 107.00
----------------------------------------------------------------
